src/averageAges.js

```diff
@@ -55,9 +75,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childs = people.filter(child => onlyWithSon
+    ? people.some(mom => child.mother === mom.name) && child.sex === 'm'
+    : people.some(mom => child.mother === mom.name));
```

Comment:
you repeat this code twice - create a variable

Reply:
done